WowPacketParser/Store/Objects/UpdateFields/LegacyImplementation/DynamicObjectData.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using WowPacketParser.Enums;
+using WowPacketParser.Misc;
+
+namespace WowPacketParser.Store.Objects.UpdateFields.LegacyImplementation
+{
+    class DynamicObjectData : IDynamicObjectData
+    {
+        private WoWObject Object { get; }
+        private Dictionary<int, UpdateField> UpdateFields => Object.UpdateFields;
+
+        public DynamicObjectData(WoWObject obj)
+        {
+            Object = obj;
+        }
+
+        private WowGuid GetGuidValue(DynamicObjectField field)
+        {
+            if (!ClientVersion.AddedInVersion(ClientType.WarlordsOfDraenor))
+            {
+                var parts = UpdateFields.GetArray<DynamicObjectField, uint>(field, 2);
+                return new WowGuid64(Utilities.MAKE_PAIR64(parts[0], parts[1]));
+            }
+            else
+            {
+                var parts = UpdateFields.GetArray<DynamicObjectField, uint>(field, 4);
+                return new WowGuid128(Utilities.MAKE_PAIR64(parts[0], parts[1]), Utilities.MAKE_PAIR64(parts[2], parts[3]));
+            }
+        }
+
+        public WowGuid Caster => GetGuidValue(DynamicObjectField.DYNAMICOBJECT_CASTER);
+
+        public int SpellXSpellVisualID => UpdateFields.GetValue<DynamicObjectField, int>(DynamicObjectField.DYNAMICOBJECT_SPELL_X_SPELL_VISUAL_ID);
+
+        public int SpellID => UpdateFields.GetValue<DynamicObjectField, int>(DynamicObjectField.DYNAMICOBJECT_SPELLID);
+
+        public float Radius => UpdateFields.GetValue<DynamicObjectField, float>(DynamicObjectField.DYNAMICOBJECT_RADIUS);
+
+        public uint CastTime => UpdateFields.GetValue<DynamicObjectField, uint>(DynamicObjectField.DYNAMICOBJECT_CASTTIME);
+
+        public byte Type => (byte)(ClientVersion.AddedInVersion(ClientVersionBuild.V8_0_1_27101)
+            ? UpdateFields.GetValue<DynamicObjectField, uint>(DynamicObjectField.DYNAMICOBJECT_TYPE)
+            : UpdateFields.GetValue<DynamicObjectField, uint>(DynamicObjectField.DYNAMICOBJECT_BYTES));
+
+    }
+    class OriginalDynamicObjectData : IDynamicObjectData
```

Comment:
The whole file does not need to be present for you. If you are curious why its duplicated like this, it's because I save the initial state of all update fields (as they were in the first create object packet received) to the database, but also need to be able to access the current state of the fields at any time while parsing, with all values updates up to this point applied. After the first create object, i make a copy of the whole update fields array, and OriginalDynamicObjectData works with it, while DynamicObjectData returns the current state of the object. This is done for all object types.